experimental/test/lib/Transforms/Speculation/commitTest.mlir

```diff
@@ -0,0 +1,34 @@
+// NOTE: Assertions have been autogenerated by utils/generate-test-checks.py
+// RUN: dynamatic-opt %s --handshake-speculation="json-path=%S/commitTest.json"
+
```

Comment:
Just realized that CHECK statements are missing the function declaration everywhere. They should be generated by the [test generator script](https://github.com/EPFL-LAP/dynamatic/blob/main/docs/Testing.md#creating-your-own-unit-tests-with-filecheck) and should be included in test files.

Reply:
I had removed some of the checks. Now all of the checks are present.